esrally/utils/git.py

```diff
@@ -48,6 +48,23 @@ def is_working_copy(src):
     return os.path.exists(src) and os.path.exists(os.path.join(src, ".git"))
 
 
+@probed
+def is_branch(src_dir, remote, identifier):
+    fetch(src_dir, remote=remote)
```

Comment:
This is very implicit, you would not expect a function called `is_branch` to affect the state by fetching. I also don'tlike that we're calling it twice in the tag/commit hash case, which can be slow when you're far away from the servers. Is there a reasonable way to avoid both issues?

Reply:
Addressed in https://github.com/elastic/rally/pull/1591/commits/af49a7002ecf21fbe7805a88acceed649d86888d